src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1103,7 +1135,7 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                 if (needDecode)
                 {
                     // Read raw target before mutating memory.
-                    rawTarget = pathBegin.GetAsciiString(ref queryEnd);
+                    rawTarget = targetBegin.GetAsciiString(ref queryEnd);
```

Comment:
IIUC there RawTarget has always been the ASCII representation. Since we ignore the authority portion of the absolute-URI, we never bother to look for a `%`